File: atcoder/abc054B_template_matching.ml
(* unihernandez22
 * https://atcoder.jp/contests/abc054/tasks/abc054_b
 * implementation
 * *)

Scanf.scanf "%d %d\n" @@ fun n m ->
    let a = Array.init n @@ fun _ -> Scanf.scanf "%s\n" @@ fun s -> s in
    let b = Array.init m @@ fun _ -> Scanf.scanf "%s\n" @@ fun s -> s in

    let valid = ref true in
    let ans = ref "No\n" in
    for i = 0 to n-m do
        for j = 0 to n-m do
            valid := true;
            for y = 0 to m-1 do
                if not (String.sub a.(i+y) j m = b.(y)) then
                    valid := false;
            done;
            if !valid then
                ans := "Yes\n";
        done;
    done;
    print_string !ans;;

File: atcoder/abc054B_template_matching.nim
# vicfred
# https://atcoder.jp/contests/abc054/tasks/abc054_b
# implementation
import strutils, sequtils

var n, m: int; (n, m) = stdin.readLine.split.map parseInt
let A = (1..n).mapIt(stdin.readLine)
let B = (1..m).mapIt(stdin.readLine)

for i in 0..n-m:
  for j in 0..n-m:
    var valid = true
    for y in 0..<m:
      if A[i+y][j..j+m-1] != B[y]:
        valid = false
    if valid:
      echo "Yes"
      quit()
echo "No"

